community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/InputRelationshipCacher.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.input;
+
+import java.io.IOException;
+
+import org.neo4j.io.fs.StoreChannel;
+
+/**
+ * Caches {@link InputRelationship} to disk using a binary format.
+ */
+public class InputRelationshipCacher extends InputEntityCacher<InputRelationship>
+{
+    private String previousType;
+
+    public InputRelationshipCacher( StoreChannel channel, StoreChannel header, int bufferSize ) throws IOException
+    {
+        super( channel, header, bufferSize, 2 );
+    }
+
+    @Override
+    protected void writeEntity( InputRelationship relationship ) throws IOException
+    {
+        // properties
+        super.writeEntity( relationship );
+
+        // id
+        if ( relationship.hasSpecificId() )
+        {
+            channel.put( (byte) 1 );
+            channel.putLong( relationship.specificId() );
+        }
+        else
+        {
+            channel.put( (byte) 0 );
+        }
+
+        // groups
+        writeGroup( relationship.startNodeGroup(), 0 );
+        writeGroup( relationship.endNodeGroup(), 1 );
+
+        // ids
+        writeValue( relationship.startNode() );
+        writeValue( relationship.endNode() );
+
+        // type
+        if ( relationship.hasTypeId() )
+        {
+            channel.put( (byte) 2 );
```

Comment:
Maybe also create constants for this marker values and use them in both cacher and reader?